### PR TITLE
Fix fieldSplit preconditioner when creating the mesh from an  image

### PR DESCRIPTION
**Pull request by fdkong**
Added missing indices for the fieldSplit preconditioner. 
    
   The current processor owns a node but the node's incident elements possibly do not live on the current processor.

### CODE REVIEW

**Comment by fdkong** [comment]
@mlesueur, fixed the issue.

File: framework/src/utils/PetscDMMoose.C [hunk 1]
```diff
@@ -550,6 +551,44 @@ DMMooseGetEmbedding_Private(DM dm, IS * embedding)
...
+                  continue;
+
+                auto node_to_elem_pair = node_to_elem_map.find(node->id());
+                is_on_current_block = false;
+                for (const auto & elem_num : node_to_elem_pair->second)
+                {
+                  // if one of incident elements belongs to a block, we consider
+                  // the node lives in the block
+                  Elem & neighbor_elem = dmm->_nl->sys().get_mesh().elem_ref(elem_num);
+                  if (neighbor_elem.subdomain_id()==b)
```
Comment (jwpeterson) [inline framework/src/utils/PetscDMMoose.C]: Spacing around `==`

**Review by jwpeterson — CHANGES_REQUESTED** [review]
We don't need a second copy of the pink kitten image.

If the new input file you added is identical to the existing FSP test except for the Mesh, consider just handling it through the `tests` file.

**Comment by YaqiWang** [comment]
The results in the failing yak test are the same as my local machine, even the convergence history. I have no idea how the change in this PR can affect the output. The gold file only have one timestep, which is expected, but the new output has two steps. @fdkong can you reproduce it on your local machine?

**Comment by fdkong** [comment]
> The results in the failing yak test are the same as my local machine, even the convergence history. I have no idea how the change in this PR can affect the output. The gold file only have one timestep, which is expected, but the new output has two steps. @fdkong can you reproduce it on your local machine?

Nope, the local machine just works fine.

**Comment by YaqiWang** [comment]
...! then let us invalidate it to see if we will see the failure again. I bet we will. We possibly need to build everything on an environment like the test box and reproduce the error first.

**Comment by fdkong** [comment]
> ...! then let us invalidate it to see if we will see the failure again. I bet we will. We possibly need to build everything on an environment like the test box and reproduce the error first.

Yak passed! Weird.

**Comment by YaqiWang** [comment]
Well, this is 💩 ...

**Comment by YaqiWang** [comment]
Wait. It could be my fault. It seems two tests are writing the same exodus file.

**Comment by YaqiWang** [comment]
A MR has been pushed in Yak for fixing the racing condition of the exodus file writing. Sorry about this.

**Comment by permcody** [comment]
image

**Comment by YaqiWang** [comment]
;-) well I cannot believe that I have fallen in this same pitfall several times and now I am creating an issue in Yak to remove the `file_base` parameter completely in the tests folder.

**Review by jwpeterson — APPROVED** [review]

**Pull request merged**